lms/djangoapps/ci_lrs/tests.py

```diff
@@ -0,0 +1,50 @@
+import random
+from unittest.mock import patch, MagicMock
+
+from django.conf import settings
+from django.utils import timezone
+from django.test import TestCase, override_settings
+
+from pymongo import MongoClient
+from pymongo.errors import ConnectionFailure, DuplicateKeyError, NetworkTimeout, OperationFailure, ServerSelectionTimeoutError
+
+from ci_lrs.utils import write_lrs_record_to_mongo
+
+
+class TestLRSTasks(TestCase):
+    def setUp(self):
+        self.mongodb = MongoClient('mongodb://mongodb:27017')['test_db']
+        self.mongodb['student_activity'].delete_many({})
+
+    def test_write_lrs_record_to_mongo(self):
+        context = {
+            'activity_time': timezone.now().isoformat(),
+            'actor': 1,
+            'verb': 'play_video',
+            'activity_object': 'myopenedx.com/urlToPage',
+            'extra_data': '{\"position\": 1}',
```

Comment:
Just going by the implementation of the lambda edx webhook - the extra_data field is parsed as json if possible, and left as a string if not. I think a couple of dataproducts are based on this assumption.

Reply:
Thanks for flagging this. That's done in the theme now.